components/ca-context.js

```diff
@@ -0,0 +1,71 @@
+define(['document-register-element'], () => {
+    /**
+     * Web component context
+     */
+    class Context extends HTMLElement {
+
+        /** @returns {string} attribute id */
+        get id() {
+            return this.getAttribute('id');
+        }
+
+        /** @param {string} value, assigned to id attrinute */
+        set id(value) {
+            this.setAttribute('id', value);
+        }
+
+        /** @returns {string} attribute name */
+        get name() {
+            return this.getAttribute('name');
+        }
+
+        /** @param {string} value, assigned to name attrinute */
+        set name(value) {
+            this.setAttribute('name', value);
+        }
+
+        /** @returns {string} user path */
+        get value() {
+            return this.getUserPath();
+        }
+
+        /** @returns {string} type */
+        get type() {
+            return 'hidden';
+        }
+
+        /**
+         * Attempts to construct a path based on the users UI
+         * We use query selector over $ so we only get the first instance of any object
+         * @returns {string} path based values
+         */
+        getUserPath() {
+            const txtProp = ('textContent' in document.createElement('i')) ? 'textContent' : 'innerText';
+            const result = [];
+
+            const nav = document.querySelector('ca-nav:not(.non-contextual) a[data-selected]');
```

Comment:
not sure about levels of browser support for :not() caniuse is a bit confusing.
http://caniuse.com/#search=%3Anot